Time series regression is not fully scikit-learn compatible

**Describe the bug**

At the moment, `MapieTimeSeriesRegressor` from `mapie.time_series_regression` is not fully scikit-learn compatible as scikit-learn's estimator check fails.

**To Reproduce**

```python
from sklearn.utils.estimator_checks import check_estimator
from mapie.time_series_regression import MapieTimeSeriesRegressor

enbpi_model = MapieTimeSeriesRegressor()
check_estimator(enbpi_model)
```

Output:

```console
AssertionError: Estimator MapieTimeSeriesRegressor should not set any attribute 
apart from parameters during init. Found attributes ['valid_methods_'].
```

**Additional context**
There are two reasons why `check_estimator` fails:
- There are non-parameter attributes set in the `__init__` method. This could be easily resolved by moving the adjustments of `cv_need_agg_function`, `valid_methods_`, and `plus_like_method` out of `__init__` into the class body just as in `MapieRegressor`.
- `partial_fit` is used for [incremental learning](https://scikit-learn.org/0.15/modules/scaling_strategies.html) in scikit-learn and in the checks it is tested whether `partial_fit` can be called even if the model has not been fitted previously. In this EnbPI implementation, however, `partial_fit` is used to update conformity scores after fitting. In my opinion, the logical thing to do would be to rename `partial_fit`.

--- a/AUTHORS.rst
+++ b/AUTHORS.rst
@@ -5,7 +5,9 @@ Credits
 Development Lead
 ----------------
 
+* Thibault Cordier <tcordier@quantmetry.com>
 * Vincent Blot <vblot@quantmetry.com>
+* Louis Lacombe <llacombe@quantmetry.com>
 * Grégoire Martinon <gmartinon@quantmetry.com>
 
 Contributors
@@ -21,8 +23,8 @@ Contributors
 * Adirtha Borgohain <adirthaborgohain@gmail.com>
 * Thomas Morzadec <tmorzadec@quantmetry.com>
 * Julien Roussel <jroussel@quantmetry.com>
-* Louis Lacombe <llacombe@quantmetry.com>
 * Arnaud Capitaine <arnaud.gc.capitaine@gmail.com>
 * Tarik Tazi <ttazi@quantmetry.com>
 * Daniel Herbst <daniel.herbst@tum.de>
+
 To be continued ...

--- a/AUTHORS.rst
+++ b/AUTHORS.rst
@@ -24,4 +24,5 @@ Contributors
 * Louis Lacombe <llacombe@quantmetry.com>
 * Arnaud Capitaine <arnaud.gc.capitaine@gmail.com>
 * Tarik Tazi <ttazi@quantmetry.com>
+* Daniel Herbst <daniel.herbst@tum.de>
 To be continued ...

